packit/distgit.py

```diff
@@ -212,10 +211,30 @@ def upload_to_lookaside_cache(self, archive_path: str) -> None:
         try:
             f.new_sources(sources=archive_path)
         except Exception as ex:
-            logger.error(f"`fedpkg new-sources` failed for some reason. "
-                         f"Either Fedora kerberos is invalid or there could be network outage.")
+            logger.error(
+                f"`fedpkg new-sources` failed for some reason. "
+                f"Either Fedora kerberos is invalid or there could be network outage."
+            )
             raise PackitException(ex)
 
+    def is_archive_on_lookaside_cache(self, archive_path: str) -> bool:
+        archive_name = os.path.basename(archive_path)
+        try:
+            res = requests.head(
+                f"https://src.fedoraproject.org/lookaside/pkgs/{self.package_name}/{archive_name}"
```

Comment:
```python
In [11]: url = f"https://src.fedoraproject.org/lookaside/pkgs/{package_name}/{archive}"

In [12]: requests.head(url)
Out[12]: <Response [301]>

In [13]: url = f"https://src.fedoraproject.org/lookaside/pkgs/{package_name}/{archive}/"

In [14]: requests.head(url)
Out[14]: <Response [200]>
```

Reply:
Is it a problem? You get `res.ok` as well. But I'll change that to receive `200`.

Reply:
it could be; I can easily imagine how we could get 301 -> redirect -> 5xx, 4xx

hence I'd indeed prefer to be in the 2xx field